src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/RSAPkcs1X509SignatureGenerator.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography.X509Certificates
+{
+    internal sealed class RSAPkcs1X509SignatureGenerator : X509SignatureGenerator
+    {
+        private readonly RSA _key;
+
+        internal RSAPkcs1X509SignatureGenerator(RSA key)
+        {
+            Debug.Assert(key != null);
+
+            _key = key;
+        }
+
+        public override byte[] SignData(byte[] data, HashAlgorithmName hashAlgorithm)
+        {
+            return _key.SignData(data, hashAlgorithm, RSASignaturePadding.Pkcs1);
+        }
+
+        protected override PublicKey BuildPublicKey()
+        {
+            RSAParameters parameters = _key.ExportParameters(false);
+
+            byte[] rsaPublicKey = DerEncoder.ConstructSequence(
+                DerEncoder.SegmentedEncodeUnsignedInteger(parameters.Modulus),
+                DerEncoder.SegmentedEncodeUnsignedInteger(parameters.Exponent));
+
+            Oid oid = new Oid(Oids.RsaRsa);
+
+            // The OID is being passed to everything here because that's what
+            // X509Certificate2.PublicKey does.
+            return new PublicKey(
+                oid,
+                // Encode the DER-NULL even though it is OPTIONAL, because everyone else does.
+                //
+                // This is due to one version of the ASN.1 not including OPTIONAL, and that was
+                // the version that got predominately implemented for RSA. Now it's convention.
+                new AsnEncodedData(oid, new byte[] { 0x05, 0x00 }),
+                new AsnEncodedData(oid, rsaPublicKey));
+        }
+
+        public override byte[] GetSignatureAlgorithmIdentifier(HashAlgorithmName hashAlgorithm)
+        {
+            string oid;
+
+            if (hashAlgorithm == HashAlgorithmName.MD5)
+            {
+                oid = Oids.RsaPkcs1Md5;
```

Comment:
Fair enough.  I can change this one to not inbox supported.  Since SHA-1 is effectively dead I guess I can cut that one, too (earlier comment) and just correct for it in the tests with a custom generator where I need to for CAPI considerations.